### PR TITLE
feat: EmbeddedTabs

### PR DESCRIPTION
**Pull request by snqb**
This PR introduces Embedded Tabs.
They are meant to be a more concise way of using Tabs in the project.

Design: at [Quartz Figma](https://www.figma.com/file/7TwZ1I9bPf6cUBgQhlvAAv/Implementation-zone?node-id=543%3A25767)

Screenshot:
image

Video Demo:
https://user-images.githubusercontent.com/19791699/167433976-eb217f0c-e8a2-4da6-8a67-85467277ddff.mp4

### CODE REVIEW

**Comment by giardiv** [comment]
Perfect 👌

**Review by giardiv — APPROVED** [review]

File: src/components/embedded-tabs/EmbeddedTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+const EmbeddedTabs: FC<Props> = ({
+  tabs,
+  onTabChange,
+  initialTab = 0,
+  sx = {},
+  ...boxProps
+}: Props) => {
+  const [active, setActive] = useState(R.clamp(0, tabs.length - 1, initialTab));
```
Comment (esonhey) [inline src/components/embedded-tabs/EmbeddedTabs.tsx]: nit: maybe `activeIndex` would be a better name

File: src/components/embedded-tabs/Tab.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const getStyles = ({ active, disabled }: Props) =>
+  ({
+    ...S.tab,
+    ...(active ? S.activeTab : {}),
+    ...(disabled ? S.disabledTab : {}),
+  } as SxStyleProp);
+
+const Tab: FC<Props> = (props) => {
+  const propagatedProps = R.pick(['disabled', 'onClick'], props);
```
Comment (esonhey) [inline src/components/embedded-tabs/Tab.tsx:22]: cool

File: src/components/embedded-tabs/EmbeddedTabs.tsx (hunk 1 above)
Comment (esonhey) [reply src/components/embedded-tabs/EmbeddedTabs.tsx]: just curious, is there any specific reason to use clamp here? you don't want to let the developer make mistake when defining the initialTabeindex?

**Review by esonhey — APPROVED** [review]
Nice Job!

File: src/components/embedded-tabs/EmbeddedTabs.tsx (hunk 1 above)
Comment (snqb) [reply src/components/embedded-tabs/EmbeddedTabs.tsx]: changed to `activeindex`.

On a second thought, the clamp is not good. I think it should fail. Removed it.

**Pull request merged**